### PR TITLE
Add nbest options and NumberFilter

### PR DESCRIPTION
**Pull request by johtani**
Add nbest_cost and nbest_examples parameter to KuromojiTokenizerFactory
Add KuromojiNumberFilterFactory

### CODE REVIEW

File: docs/plugins/analysis-kuromoji.asciidoc [hunk 1]
```diff
@@ -452,3 +473,48 @@ The above request returns:
...
+PUT kuromoji_sample
+{
+  "settings": {
+    "index": {
+      "analysis": {
+        "analyzer": {
+          "my_analyzer": {
+            "tokenizer": "kuromoji_tokenizer",
+            "filter": [
+              "kuromoji_nymber"
```
Comment (jpountz) [inline docs/plugins/analysis-kuromoji.asciidoc]: s/nymber/number/

File: plugins/analysis-kuromoji/src/main/java/org/elasticsearch/index/analysis/KuromojiTokenizerFactory.java [hunk 2]
```diff
@@ -47,6 +51,8 @@ public KuromojiTokenizerFactory(IndexSettings indexSettings, Environment env, St
         mode = getMode(settings);
         userDictionary = getUserDictionary(env, settings);
         discartPunctuation = settings.getAsBoolean("discard_punctuation", true);
+        nBestCost = settings.getAsInt(NBEST_COST, -1);
+        nBestExamples = settings.get(NBEST_EXAMPLES);
```
Comment (jpountz) [inline plugins/analysis-kuromoji/src/main/java/org/elasticsearch/index/analysis/KuromojiTokenizerFactory.java]: should we fail if both nbest_cost and nbest_examples are set?

Comment (johtani) [inline plugins/analysis-kuromoji/src/main/java/org/elasticsearch/index/analysis/KuromojiTokenizerFactory.java]: Ah, No. Lucene supports that both parameters are set.
https://github.com/apache/lucene-solr/blob/master/lucene/analysis/kuromoji/src/java/org/apache/lucene/analysis/ja/JapaneseTokenizerFactory.java#L74

I will add test case and documentation.

Comment (jpountz) [inline plugins/analysis-kuromoji/src/main/java/org/elasticsearch/index/analysis/KuromojiTokenizerFactory.java]: OK

Comment (johtani) [inline plugins/analysis-kuromoji/src/main/java/org/elasticsearch/index/analysis/KuromojiTokenizerFactory.java]: Added documentation and test.

**Comment by jpountz** [comment]
LGTM

**Pull request closed without merging**